### PR TITLE
CCD-739 Implement new instrument CALPARS and populate initial specs for known Steps and Pipelines

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [CCD-739](https://jira.stsci.edu/browse/CCD-739)

Create the new instrument `calpars`. This pseudo-instrument will house all the JWST step parameter references files used by the `jwst.stpipe.Step` classes. The initial specs for the currently known Steps are populated.

### CODE REVIEW

File: crds/jwst/tpns/calpars_all.tpn [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Template file used by certify to check reference files
+# Some fields may be abbreviated to their first character:
+#
+# keytype = (Header|Group|Column)
+# datatype = (Integer|Real|Logical|Double|Character)
+# presence = (Optional|Required)
+#
+# NAME  KEYTYPE  DATATYPE   PRESENCE    VALUES
+#----------------------------------------------------------
+META.INSTRUMENT.NAME        H   C   R    CALPARS
```
Comment (eslavich) [inline crds/jwst/tpns/calpars_all.tpn:10]: Is it worth adding any of the other common attributes?  Or is the file pretty well covered by the model schema?

File: documentation/crds_users_guide/source/file_submissions.rst [hunk 2]
```diff
@@ -512,18 +512,22 @@ Manual .imap update
```
Comment (eslavich) [inline documentation/crds_users_guide/source/file_submissions.rst:512]: Thanks for this!

File: install [hunk 3]
```diff
@@ -2,20 +2,26 @@
...
-
 if [ "$1" = "--skip-types" ]; then
     shift;
     dont_skip_types=0
 else
     dont_skip_types=1
 fi
 
+dev_install=0
+if [ "$1" == "--dev" ]; then
```
Comment (eslavich) [inline install:13]: This doesn't work if the script is called with the flags in order `--dev --skip-types`?

File: install [hunk 4]
```diff
@@ -49,18 +55,24 @@ if [ $dont_skip_types = 1 ]; then
     python -m crds.tobs
 fi
 
-pip install .
+if [ $dev_install = 0 ]; then
+    echo "Doing setup install..."
+    pip install .
```
Comment (eslavich) [inline install:60]: Where does the pip install occur when `$dev_install = 1`?

File: crds/jwst/specs/calpars_pars-alignrefsstep.rmap [hunk 5]
```diff
@@ -0,0 +1,17 @@
+header = {
```
Comment (eslavich) [inline crds/jwst/specs/calpars_pars-alignrefsstep.rmap:1]: Woo wee I hope you get paid by the rmap!

File: crds/jwst/tpns/calpars_all.tpn (hunk 1 above)
Comment (stscieisenhamer) [reply crds/jwst/tpns/calpars_all.tpn:10]: Yes. In fact, this should probably include each instruments `all` tpns. Except for instrument, all other parameters will be the instrument-specific ones.

File: install (hunk 4 above)
Comment (stscieisenhamer) [reply install:60]: This was the main goal. If `crds` had been originally installed with `pip -e`, there is no desire to have things installed again.

That being said, and some enlightenment from the above comment, what may really be needed is another script that does just the checksum update (and whatever other operation), and restore `install` back to what it was, except for calling the new script.

File: crds/jwst/specs/calpars_pars-alignrefsstep.rmap (hunk 5 above)
Comment (stscieisenhamer) [reply crds/jwst/specs/calpars_pars-alignrefsstep.rmap:1]: 🤑

**Review by jaytmiller — APPROVED** [review]
This looks good to me.  In addition to testing the submission and bestrefs functionality be sure to check "crds sync" and maybe the offsite dynamic cache functionality to make sure pipeline cache updates are working.  Someday when repro becomes a more urgent subject,  how these files affect repro will need to be considered,  but that's a topic for another day.

**Comment by stscieisenhamer** [comment]
Do to reasons as documented in [crds-381](https://jira.stsci.edu/browse/CRDS-381), this approach has been deemed not optimal. Closing out the PR.

**Pull request closed without merging**